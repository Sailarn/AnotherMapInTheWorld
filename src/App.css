.autocomplete-dropdown-container{
    position: absolute;
    z-index: 99999;
    box-shadow: 0 2px 5px 0 rgba(0, 0, 0, 0.16), 0 2px 10px 0 rgba(0, 0, 0, 0.12);
    border-radius: .25rem;
}
.md-form{
    margin-bottom: 0;
}
.suggestion-item{
    padding: 5px;
}
.suggestion-item--active{
    padding: 5px;
    font-weight: 700;
}
.delete-item-box{
    width: 50px;
    border: 1px solid rgba(0,0,0,.125);
    border-left: none;
    display: flex;
    justify-content: center;
    align-items: center;
    transition: all ease 5ms;
}
.delete-item-box:hover{
    background-color: rgba(0,0,0,.05);
    cursor: pointer;
}
.delete-item-btn{
    color: #212529;
}
.scroll-card{
    overflow-y: scroll;
}
.index-class{
    margin: 0;
    border-right: none;
    border-left: 1px solid rgba(0,0,0,.125);
}
.list-item-box{
    margin-bottom: 5px;
    display: flex;
    justify-content: center;
    align-items: stretch;
}
.list-item-box:hover{
    cursor: pointer;
}
.preloader{
    position: absolute;
    top: 0;
    left: 0;
    z-index: 99999;
    background-color: white;
    width: 100vw;
    height: 100vh;
    display: flex;
    justify-content: center;
    align-items: center;
}
.switch-input {
    display: none;
}
.switch-label {
    position: relative;
    display: inline-block;
    min-width: 112px;
    cursor: pointer;
    font-weight: 500;
    text-align: left;
    margin: 16px;
    padding: 16px 0 16px 44px;
}
.switch-label:before, .switch-label:after {
    content: "";
    position: absolute;
    margin: 0;
    outline: 0;
    top: 50%;
    -ms-transform: translate(0, -50%);
    -webkit-transform: translate(0, -50%);
    transform: translate(0, -50%);
    -webkit-transition: all 0.3s ease;
    transition: all 0.3s ease;
}
.switch-label:before {
    left: 1px;
    width: 34px;
    height: 14px;
    background-color: #9E9E9E;
    border-radius: 8px;
}
.switch-label:after {
    left: 0;
    width: 20px;
    height: 20px;
    background-color: #FAFAFA;
    border-radius: 50%;
    box-shadow: 0 3px 1px -2px rgba(0, 0, 0, 0.14), 0 2px 2px 0 rgba(0, 0, 0, 0.098), 0 1px 5px 0 rgba(0, 0, 0, 0.084);
}
.switch-label .toggle--on {
    display: none;
}
.switch-label .toggle--off {
    display: inline-block;
}
.switch-input:checked + .switch-label:before {
    background-color: #A5D6A7;
}
.switch-input:checked + .switch-label:after {
    background-color: #4CAF50;
    -ms-transform: translate(80%, -50%);
    -webkit-transform: translate(80%, -50%);
    transform: translate(80%, -50%);
}
.switch-input:checked + .switch-label .toggle--on {
    display: inline-block;
}
.switch-input:checked + .switch-label .toggle--off {
    display: none;
}
.settings-wrapper{
    position: absolute;
    z-index: 1;
    width: 100%;
    height: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
}
.settings{
    position: relative;
    width: 100%;
    max-height: 700px;
    overflow-y: scroll;
    background-color: rgba(255, 255, 255, 0.95);
}
.hide{
    right: -110%;
}
.z0{
    z-index: 0 !important;
}
.list-group-item{
    margin-bottom: 0;
}
.toaster{
    z-index: 99999;
    max-height: 200px;
}
@media screen and (max-width: 1024px) {
    .row{
        justify-content: center;
        align-items: center;
        flex-direction: column-reverse;
    }
    .col-6{
        max-width: 100% !important;
        margin-bottom: 20px;
    }
}
@media screen and (max-width: 380px) {
    .options{
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
    }
}
@media screen and (max-height: 700px) {
    .settings{
        max-height: 400px;
    }
}
