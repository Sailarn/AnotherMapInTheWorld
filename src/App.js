import React, {Component} from 'react';
import './App.css';
import GoogleMap from './Components/Map/Map'
import {
    MDBNavbar,
    MDBNavbarBrand,
    MDBNavbarNav,
    MDBNavItem,
    MDBNavbarToggler,
    MDBCollapse,
    MDBBtn,
    MDBRow,
    MDBContainer,
    MDBInput,
    MDBCard,
    MDBCardBody,
    MDBCardTitle,
    MDBCardText,
    MDBCol,
    MDBListGroup, MDBListGroupItem
} from "mdbreact";
import {fromWaypoint} from "./store/actions/mapAction";
import {connect} from "react-redux";
import PlacesAutocomplete, {
    geocodeByAddress,
    getLatLng,
} from 'react-places-autocomplete';

/* global google */
class App extends Component {
    state = {
        isOpen: false,
        valueFrom: '',
        fromLatLng: '',
        waypoints: [],
        polyline: [],
        markers: [],
        addresses: []
    };

    componentDidUpdate() {

    }

    setPoint = (latLng, address) => {

        // var infowindow = new google.maps.InfoWindow({
        //     content: address
        // });
        // marker.addListener('click', () => {
        //     //infowindow.open(this.props.google, marker);
        // });
        let waypoints = this.state.waypoints;
        let polyline = this.state.polyline;
        let markers = this.state.markers;
        if (waypoints.length > 1) {
            polyline.setMap(null);
        }
        waypoints.push(latLng);

        let marker = new google.maps.Marker({
            position: latLng,
            map: this.props.google
        });
        markers.push(marker);
        polyline = new google.maps.Polyline({
            path: waypoints,
            geodesic: true,
            strokeColor: '#FF0000',
            strokeOpacity: 1.0,
            strokeWeight: 2
        });
        polyline.setMap(this.props.google);
        this.setState({
            waypoints,
            polyline,
            markers
        })
        google.maps.event.addListener(marker, 'click', (event) => {
            this.removePoint(marker);
        });

    }

    removePoint(marker) {
        let markers = this.state.markers;
        let waypoints = this.state.waypoints;
        for (var i = 0; i < markers.length; i++) {
            console.log(marker, markers)
            if (markers[i] === marker) {
                markers[i].setMap(null);
                markers.splice(i, 1);
                waypoints.splice(i, 1);
                this.state.polyline.getPath().removeAt(i);
                this.setState({
                    markers,
                    waypoints
                })
            }
        }
    }
    deleteMarker = (event) =>{
        let markers = this.state.markers;
        let marker = markers[event];
        let addresses = this.state.addresses;
        let waypoints = this.state.waypoints;
        for (var i = 0; i < markers.length; i++) {
            if (markers[i] === marker) {
                markers[i].setMap(null);
                markers.splice(i, 1);
                waypoints.splice(i, 1);
                addresses.splice(i, 1);
                this.state.polyline.getPath().removeAt(i);

                this.setState({
                    markers,
                    waypoints,
                    addresses
                })
            }
        }
    }

    toggleCollapse = () => {
        this.setState({isOpen: !this.state.isOpen});
    }
    inputOnChange = (event) => {
        this.setState({
            [event.target.name]: event.target.value
        })
        console.log(google)
    }
    enterPress = (event) => {
        // //if(this.state.waypoints.length >= 2) {
        // console.log('yez')
        // let request;
        // var directionsService = new google.maps.DirectionsService;
        // var directionsDisplay = new google.maps.DirectionsRenderer;
        // if (this.state.waypoints.length > 0) {
        //     directionsDisplay.set('directions', null);
        // }
        //
        // directionsDisplay.setOptions({suppressMarkers: false, draggable: true});
        //
        // directionsDisplay.setMap(this.props.google);
        //
        // //var waypts = [{location: {lat: 49.9935, lng: 36.230383000000074}, stopover: true}];
        // //{location: {lat: 49.9935, lng: 36.230383000000074}, stopover: true}
        // if (this.state.waypoints.length >= 1) {
        //     console.log('waypoints')
        //     let waypts = this.state.waypoints;
        //     let waypoints = [];
        //     for (let item of waypts) {
        //         waypoints.push({location: item, stopover: true});
        //     }
        //     console.log(waypoints)
        //     // let value = event.target.value;
        //     // console.log(event.target.value)
        //
        //     // console.log(this.props.fromValue)
        //     //50.746957, 25.325646 {lat: 50.746957, lng: 25.325646}
        //     request = {
        //         origin: new google.maps.LatLng(this.state.fromLatLng), //точка старта
        //         destination: new google.maps.LatLng(this.state.toLatLng), //точка финиша
        //         waypoints: waypoints,
        //         optimizeWaypoints: true,
        //         travelMode: google.maps.DirectionsTravelMode.WALKING //режим прокладки маршрута
        //     };
        // } else {
        //     request = {
        //         origin: new google.maps.LatLng(this.state.fromLatLng), //точка старта
        //         destination: new google.maps.LatLng(this.state.toLatLng), //точка финиша
        //         optimizeWaypoints: true,
        //         travelMode: google.maps.DirectionsTravelMode.WALKING //режим прокладки маршрута
        //     };
        // }
        // console.log(this.state.fromLatLng, this.state.toLatLng)
        // directionsService.route(request, function (response, status) {
        //     if (status === window.google.maps.DirectionsStatus.OK) {
        //         directionsDisplay.setDirections(response);
        //     }
        // });
        // directionsDisplay.setMap(this.props.google);
        // //}
        // //50.746957,25.325646
    }

    addListItem =() => {
        return Object.keys(this.state.addresses).map((pos, index) => {
            let marker = this.state.markers[pos];
            let address = this.state.addresses[pos];
            return (
                <MDBListGroupItem
                    onClick={this.deleteMarker.bind(null, pos)}
                    hover
                    key={index}
                >{address}</MDBListGroupItem>
            );
        });
    }

    handleSelect = address => {
        let addresses = this.state.addresses;
        addresses.push(address);
        this.setState({
            addresses
        })
        geocodeByAddress(address)
            .then(results => getLatLng(results[0]))
            .then(latLng => {
                this.setPoint(latLng, address)
                this.setState({
                    valueFrom: ''
                })
            })
            .catch(error => console.error('Error', error));
    };

    render() {
        return (
            <div className="App">
                <MDBNavbar color="indigo" dark expand="md">
                    <MDBNavbarBrand>
                        <strong className="white-text">RouteFinder</strong>
                    </MDBNavbarBrand>
                    <MDBNavbarToggler onClick={this.toggleCollapse}/>
                    <MDBCollapse id="navbarCollapse3" isOpen={this.state.isOpen} navbar>
                        <MDBNavbarNav left>
                            <MDBNavItem active>
                                <MDBBtn size="sm">Create Route</MDBBtn>
                            </MDBNavItem>
                        </MDBNavbarNav>
                    </MDBCollapse>
                </MDBNavbar>
                <MDBContainer fluid style={{marginTop: '20px'}}>
                    <MDBRow around style={{display: 'flex', flexWrap: 'wrap'}}>
                        <MDBCol size="4">
                            <MDBCard>
                                <MDBCardBody>
                                    <MDBCardTitle>Card title</MDBCardTitle>
                                    <MDBCardText>
                                        Some quick example text to build on the card title and make
                                        up the bulk of the card&apos;s content.
                                    </MDBCardText>
                                    <PlacesAutocomplete
                                        value={this.state.valueFrom}
                                        onChange={valueFrom => this.setState({valueFrom})}
                                        onSelect={this.handleSelect}
                                    >
                                        {({getInputProps, suggestions, getSuggestionItemProps, loading}) => (
                                            <div>
                                                <MDBInput
                                                    hint="From"
                                                    id="from_value"
                                                    name="valueFrom"
                                                    //value={this.state.fromValue}
                                                    {...getInputProps({
                                                        placeholder: 'Search Places ...',
                                                        className: 'location-search-input',
                                                    })}
                                                />
                                                <div className="autocomplete-dropdown-container">
                                                    {loading && <div>Loading...</div>}
                                                    {suggestions.map(suggestion => {
                                                        const className = suggestion.active
                                                            ? 'suggestion-item--active'
                                                            : 'suggestion-item';
                                                        // inline style for demonstration purpose
                                                        const style = suggestion.active
                                                            ? {backgroundColor: '#fafafa', cursor: 'pointer'}
                                                            : {backgroundColor: '#ffffff', cursor: 'pointer'};
                                                        return (
                                                            <div
                                                                {...getSuggestionItemProps(suggestion, {
                                                                    className,
                                                                    style,
                                                                })}
                                                            >
                                                                <span>{suggestion.description}</span>
                                                            </div>
                                                        );
                                                    })}
                                                </div>
                                            </div>
                                        )}
                                    </PlacesAutocomplete>
                                    <MDBBtn size="md" onClick={this.enterPress}>Create Route</MDBBtn>
                                    <MDBBtn size="md" onClick={() => {
                                        let polyline = this.state.polyline.setMap(null);
                                        this.setState({
                                            polyline
                                        })
                                    }}>Test</MDBBtn>
                                    <MDBListGroup style={{marginTop: '25px'}}>
                                        {this.state.addresses.length >= 1 ? this.addListItem() : false}
                                    </MDBListGroup>
                                </MDBCardBody>
                            </MDBCard>
                        </MDBCol>
                        <MDBCol size="8">
                            <MDBCard>
                                <MDBCardBody>
                                    <MDBCardTitle>Card title</MDBCardTitle>
                                    <GoogleMap/>
                                </MDBCardBody>
                            </MDBCard>
                        </MDBCol>
                    </MDBRow>
                </MDBContainer>
            </div>
        );
    }
}

function mapDispatchToProps(dispatch) {
    return {
        fromWaypoint: waypoint => dispatch(fromWaypoint(waypoint))
    };
}

function mapStateToProps(state) {
    return {
        google: state.map.google,
        fromValue: state.map.fromWaypoint
    };
}

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(App);
