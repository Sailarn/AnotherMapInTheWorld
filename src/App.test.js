import React from 'react';
import { mount, shallow, render } from 'enzyme';
import renderer from 'react-test-renderer';
import App from './App';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {BrowserRouter} from 'react-router-dom';
import {createStore, compose, applyMiddleware} from 'redux';
import {Provider} from 'react-redux';
import rootReducer from './store/reducers/rootReducer';
import thunk from 'redux-thunk';

configure({ adapter: new Adapter() });

const composeEnhancers =
    typeof window === 'object' &&
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?
        window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
        }) : compose;

const store = createStore(
    rootReducer,
    composeEnhancers(
        applyMiddleware(thunk)
    )
);
const application = (
    <Provider store={store}>
        <BrowserRouter>
            <App />
        </BrowserRouter>
    </Provider>
)

describe('MyComponent', () => {
    const app = renderer.create(application);
    it('Render App component correctly', () => {
        let tree = app.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
