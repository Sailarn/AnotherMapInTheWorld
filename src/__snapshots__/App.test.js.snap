// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent Render App component correctly 1`] = `
<div
  className="App"
>
  <div
    className="preloader"
  >
    <div
      className="spinner-grow text-info"
      role="status"
    />
  </div>
  <nav
    className="navbar navbar-dark navbar-expand-md elegant-color-dark"
    role="navigation"
  >
    <div
      className="navbar-brand"
    >
      <strong
        className="white-text"
      >
        RouteFinder
      </strong>
    </div>
    <button
      className="btn btn-amber btn-sm Ripple-parent toggleOptions"
      onClick={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      Toggle options
      <div
        className="Ripple "
        style={
          Object {
            "height": "0px",
            "left": "0px",
            "top": "0px",
            "width": "0px",
          }
        }
      />
    </button>
  </nav>
  <div
    className="container-fluid"
  >
    <div />
    <div
      className="row justify-content-around"
    >
      <div
        className="col-12 settings-wrapper z0"
      >
        <div
          className="card d-none"
        >
          <div
            className="card-body"
          >
            <h4
              className="card-title"
            >
              Settings
            </h4>
            <p
              className="card-text"
            >
              You can choose variant with arrows and press Enter to add a waypoint.
            </p>
            <div
              className="options"
            >
              <button
                className="btn btn-default btn-sm Ripple-parent"
                onClick={[Function]}
                onMouseUp={[Function]}
                onTouchStart={[Function]}
                type="button"
              >
                Create Route
                <div
                  className="Ripple "
                  style={
                    Object {
                      "height": "0px",
                      "left": "0px",
                      "top": "0px",
                      "width": "0px",
                    }
                  }
                />
              </button>
              <input
                className="switch-input"
                id="id-name--1"
                name="edit-mode"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="switch-label"
                htmlFor="id-name--1"
              >
                Edit mode 
                <span
                  className="toggle--on"
                >
                  On
                </span>
                <span
                  className="toggle--off"
                >
                  Off
                </span>
              </label>
              <div
                className="btn-group-sm dropdown"
                onClick={[Function]}
                onKeyDown={[Function]}
                style={
                  Object {
                    "display": "inline-flex",
                  }
                }
              >
                <button
                  aria-expanded={false}
                  aria-haspopup={true}
                  className="btn btn-mdb-color Ripple-parent dropdown-toggle"
                  defaultValue=""
                  onClick={[Function]}
                  onMouseUp={[Function]}
                  onTouchStart={[Function]}
                  type="button"
                >
                  Travel Mode
                  <div
                    className="Ripple "
                    style={
                      Object {
                        "height": "0px",
                        "left": "0px",
                        "top": "0px",
                        "width": "0px",
                      }
                    }
                  />
                </button>
                <div
                  aria-hidden={true}
                  className="dropdown-menu"
                  role="menu"
                  tabIndex="-1"
                >
                  <button
                    className="dropdown-item"
                    onClick={[Function]}
                    tabIndex="0"
                    type="button"
                    value="DRIVING"
                  >
                    Driving
                  </button>
                  <button
                    className="dropdown-item"
                    onClick={[Function]}
                    tabIndex="0"
                    type="button"
                    value="BICYCLING"
                  >
                    Bicycling
                  </button>
                  <button
                    className="dropdown-item"
                    onClick={[Function]}
                    tabIndex="0"
                    type="button"
                    value="TRANSIT"
                  >
                    Transit
                  </button>
                  <button
                    className="dropdown-item"
                    onClick={[Function]}
                    tabIndex="0"
                    type="button"
                    value="WALKING"
                  >
                    Walking
                  </button>
                </div>
              </div>
            </div>
            <h3>
              Waypoints
            </h3>
            <ul
              className="list-group"
              style={
                Object {
                  "marginTop": "25px",
                }
              }
            >
              No waypoints created
            </ul>
            <h3
              style={
                Object {
                  "marginTop": "10px",
                }
              }
            >
              Routes
            </h3>
            <ul
              className="list-group"
              style={
                Object {
                  "marginTop": "25px",
                }
              }
            >
              No routes created
            </ul>
          </div>
        </div>
      </div>
      <div
        id="map"
      />
    </div>
  </div>
</div>
`;
